pandas/core/dtypes/concat.py

```diff
@@ -491,3 +491,30 @@ def convert_sparse(x, axis):
         # coerce to object if needed
         result = result.astype('object')
     return result
+
+
+def _concat_indexes(to_concat, default=None):
+    """
```

Comment:
so this is good, but can you use ``_concat_compat`` which basically does all of the combination logic, rather than have it live IN the Index classes themselves? (you would simply wrap it to return an ``Index``, rather than a Series as is done elsewhere).

My goal is to centralize all of this code, and NOT decentralize it to the Index whenever possible.